Sources/YNetwork/NetworkManager/Progress/FileProgress.swift

```diff
@@ -57,10 +71,17 @@ internal class FileProgress: NSObject {
     /// - Parameters:
     ///   - cancelable: optional cancelable task
     ///   - progress: optional progress handler
-    func register(_ cancelable: Cancelable?, progress: ProgressHandler?) {
+    func register(
+        _ cancelable: Cancelable?,
+        progress: ProgressHandler?,
+        completion: CompletionHandler? = nil
+    ) {
         guard let task = cancelable as? URLSessionTask,
               let progress = progress else { return }
         progressHandlersByTaskID[task.taskIdentifier] = progress
+        if let completion = completion {
+            completionHandlerByTaskID[task.taskIdentifier] = completion
+        }
```

Comment:
if progress handler is passed nil while calling this method for some reason, then completion handler also will not get called. Can you please handle this scenario.

Reply:
I refactored the register APIs related to upload handling and added more documentation comments in the code. For this particular issue, the upload completion handler registration was separated from the progress handler registration.